### PR TITLE
Engineering Gripper for Engineering-focused Cyborgs

### PR DESCRIPTION
**Pull request by Runian**
# Document the changes in your pull request
This ports three different PRs from Citadel with a few changes:
Citadel-Station-13/Citadel-Station-13#9619
Citadel-Station-13/Citadel-Station-13#12780
Citadel-Station-13/Citadel-Station-13#15103

Cyborgs can longer drag n' drop circuit boards and stock parts into machine frames; same goes for computer frames.

Engineering (and Saboteur) Cyborgs now have a gripper that can be used to carry and use circuit boards, stock parts and more.
These are the types that they can use along with a rationale:
- /obj/item/circuitboard 
They could always do this even without a gripper. It is now more convenient than dragging a board around.
- /obj/item/electronics
This is an alternative to the electroadaptive pseudocircuit that they have. With this, they can go manually find the electronic and use it. This also means that they repair a emagged/broken airlock without needing to use a RCD.
- /obj/item/wallframe
They could always do this even without a gripper by moving ontop of the wallframe and then using a screwdriver on it.
- /obj/item/stock_parts
They could always do this even without a gripper. It is now more convenient than dragging a stock part around.

AIs and Cyborgs now can eject cells from a cell charger.

# Spriting
Ingame:
image

# Wiki Documentation
Add this to [Cyborg's Engineering](https://wiki.yogstation.net/wiki/Cyborg#Engineering) and [Cyborg's Saboteur](https://wiki.yogstation.net/wiki/Cyborg#Syndicate:_Saboteur). Make sure to mention this is the only way for them to use circuit boards and stock parts.

# Changelog
:cl:  
rscdel: Cyborgs can no longer drag n' drop circuit boards and stock parts into machine frames.
rscdel: Cyborgs can no longer drag n' drop circuit boards into computer frames.
rscadd: Cyborgs with an engineering-focus now have a gripper that can be used for engineering purposes. 
rscadd: The gripper can pick up and use items such as circuit boards, stock parts, and electronics. Some exceptions apply.
rscadd: AI and Cyborgs can now eject a cell from a cell charger.
imageadd: Gripper sprite.
/:cl:

### CODE REVIEW

**Comment by azzzertyy** [comment]
as mentioned on discord, allows a borg to scoop up an ai upload board and then Delete it

**Comment by Runian** [comment]
> as mentioned on discord, allows a borg to scoop up an ai upload board and then Delete it

I'll fix it once I figure out how to.

**Comment by iamthesilly** [comment]
Would be nice if it could hold stuff like airlock electronics too

**Comment by Runian** [comment]
> Would be nice if it could hold stuff like airlock electronics too

That is for [another](https://github.com/Citadel-Station-13/Citadel-Station-13/pull/12780) [time](https://github.com/Citadel-Station-13/Citadel-Station-13/pull/15103). Just want to get this through the front door.

EDIT: Decided to port the rest of the things, except heavily reduce the whitelisted item pool. Airlock electronics can be used now.

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -756,3 +756,73 @@
...
 	. = ..()
 	hud = new /obj/item/clothing/glasses/hud/security(src)
+
+/**********************************************************************
+						Grippers
+***********************************************************************/
+
+/obj/item/weapon/gripper
+	name = "circuit gripper"
+	desc = "A simple grasping tool for inserting circuitboards into machinary."
```
Comment (hedgehog1029) [inline code/game/objects/items/robot/robot_items.dm]: machinary

suggested: `"A simple grasping tool for inserting circuit boards into machinery."`

**Comment by ju45he** [comment]
What if, it could pickup a holofan? Or make the tiny fans on the ground in some places to block air movement able to be crafted, maybe nerf crafted ones so they only function with power. Please I'm begging you.

**Comment by Runian** [comment]
> What if, it could pickup a holofan? Or make the tiny fans on the ground in some places to block air movement able to be crafted, maybe nerf crafted ones so they only function with power. Please I'm begging you.

That can and should be a different PR.

**Review by cuackles — APPROVED** [review]

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**